packages/components/src/DataTable/Item/DataTableRow.tsx

```diff
@@ -76,7 +76,12 @@ const DataTableRowLayout = forwardRef(
       if (columnsDisplayState && !columnsDisplayState[index]) {
         return null
       } else if (isValidElement(child)) {
-        return cloneElement(child, { size: getColumnSize(index) })
+        const size = getColumnSize(index)
+        const cellProps =
+          index === 0
```

Comment:
Rather than using `index === 0` use `isHeaderRow` instead:

```suggestion
          isHeaderRow 
```